87-добавление-данных-в-профилелюбом-имя-фамилия-затирает-дефолтный-ав…

neuromanzer opened the pull request:
…атар-смена-пароля-у-того-же-юзера-затирает-ранее-занесенные-в

переписана логика апдейта юзера в профиле

neuromanzer committed:
87-добавление-данных-в-профилелюбом-имя-фамилия-затирает-дефолтный-аватар-смена-пароля-у-того-же-юзера-затирает-ранее-занесенные-в
переписана логика апдейта юзера в профиле

neuromanzer committed:
87-добавление-данных-в-профилелюбом-имя-фамилия-затирает-дефолтный-аватар-смена-пароля-у-того-же-юзера-затирает-ранее-занесенные-в
переписаны эндпоинты для ролей ADMIN & MANAGER
добвалено изменение почты ролей ADMIN & MANAGER

neuromanzer committed:
87-добавление-данных-в-профилелюбом-имя-фамилия-затирает-дефолтный-аватар-смена-пароля-у-того-же-юзера-затирает-ранее-занесенные-в
починено добавление аватарок

neuromanzer committed:
87-добавление-данных-в-профилелюбом-имя-фамилия-затирает-дефолтный-аватар-смена-пароля-у-того-же-юзера-затирает-ранее-занесенные-в
исправлено отображение акций

dshishkaryan commented on src/main/java/com/jm/online_store/controller/rest/AuthorityRestController.java:
```diff
@@ -0,0 +1,39 @@
+    /**
+     * REST-контролллер для ролей ADMIN & MANAGER
+     */
+
```
лишня строка

neuromanzer replied:
+

dshishkaryan commented on src/main/java/com/jm/online_store/controller/simple/AllUsersController.java:
```diff
@@ -0,0 +1,38 @@
+
+@Controller
+@RequestMapping(value = "/users")
+//@RequiredArgsConstructor
```
если не нужен то лучше убрать

neuromanzer replied:
+

dshishkaryan commented on src/main/java/com/jm/online_store/controller/simple/AllUsersController.java:
```diff
@@ -0,0 +1,38 @@
+    private UserService userService;
+
+    @PostMapping("/uploadImage")
+    @ResponseBody
```
судя по методам это у тебя рест контроллер получается. Его лучше перенести в пакет рест и дооформить что бы ResponseEntity возвращал

dshishkaryan commented on src/main/java/com/jm/online_store/controller/simple/AuthorityController.java:
```diff
@@ -4,29 +4,26 @@
+    /**
+     * Контролллер для ролей ADMIN MANAGER
+     */
 
```
строка лишняя

neuromanzer replied:
+

dshishkaryan commented on src/main/java/com/jm/online_store/controller/simple/AuthorityController.java:
```diff
@@ -42,15 +39,14 @@ public String getPersonalInfo(Model model, Authentication auth) {
+        return "profileAuthority";
     }
 
     @PostMapping("/profile")
```
я бы тут поменял бы на @PutMapping т.к. не новый объект а update

neuromanzer replied:
с @PutMapping кидает ошибку

dshishkaryan replied:
Ну ошибка скорей всего из за того что с фронта POST а тут поменял на PUT

dshishkaryan commented on src/main/java/com/jm/online_store/controller/simple/AuthorityController.java:
```diff
@@ -63,38 +59,10 @@ public String changePassword(Authentication auth, Model model,
                                  @RequestParam String newPassword) {
         User user = (User) auth.getPrincipal();
-        if (!passwordEncoder.matches(oldPassword, user.getPassword())) {
+        if (!userService.changePassword(user.getId(), oldPassword, newPassword)) {
```
всю логику лучше перенести в сервисный слой

dshishkaryan commented on src/main/java/com/jm/online_store/controller/simple/AuthorityController.java:
```diff
@@ -63,38 +59,10 @@ public String changePassword(Authentication auth, Model model,
-        if (!passwordEncoder.matches(oldPassword, user.getPassword())) {
+        if (!userService.changePassword(user.getId(), oldPassword, newPassword)) {
             model.addAttribute("message", "Pls, double check previous password!");
-
```
строка пустая

dshishkaryan commented on src/main/java/com/jm/online_store/controller/simple/AuthorityController.java:
```diff
@@ -63,38 +59,10 @@ public String changePassword(Authentication auth, Model model,
         }
-        user.setPassword(passwordEncoder.encode(newPassword));
-        userService.updateUser(user);
-
```
еще строка

dshishkaryan commented on src/main/java/com/jm/online_store/controller/simple/AuthorityController.java:
```diff
@@ -63,38 +59,10 @@ public String changePassword(Authentication auth, Model model,
                                  @RequestParam String newPassword) {
         User user = (User) auth.getPrincipal();
-        if (!passwordEncoder.matches(oldPassword, user.getPassword())) {
+        if (!userService.changePassword(user.getId(), oldPassword, newPassword)) {
```
Всю логику лучше в сервисный слой перенести если возможно

neuromanzer replied:
тут логика как раз в сервисе

neuromanzer committed:
87-добавление-данных-в-профилелюбом-имя-фамилия-затирает-дефолтный-аватар-смена-пароля-у-того-же-юзера-затирает-ранее-занесенные-в
поправки по гиту

neuromanzer committed:
87-добавление-данных-в-профилелюбом-имя-фамилия-затирает-дефолтный-аватар-смена-пароля-у-того-же-юзера-затирает-ранее-занесенные-в

при смене имейла перекидывает на адрес в зависимости от роли пользователя

neuromanzer committed:
87-добавление-данных-в-профилелюбом-имя-фамилия-затирает-дефолтный-аватар-смена-пароля-у-того-же-юзера-затирает-ранее-занесенные-в
починено добавление дефолтного аватара при создании нового пользователя админом

neuromanzer committed:
87-добавление-данных-в-профилелюбом-имя-фамилия-затирает-дефолтный-аватар-смена-пароля-у-того-же-юзера-затирает-ранее-занесенные-в
починено добавление дефолтного аватара при создании нового пользователя админом
подгрузка аватара в профиле, не поностью рабочая версия

neuromanzer committed:
87-добавление-данных-в-профилелюбом-имя-фамилия-затирает-дефолтный-аватар-смена-пароля-у-того-же-юзера-затирает-ранее-занесенные-в

при смене аватара с похожим именем старый не затирается
при загрузке аватара генерируется уникальное имя

dshishkaryan commented on src/main/java/com/jm/online_store/controller/simple/AllUsersController.java:
```diff
@@ -0,0 +1,38 @@
+    private final UserService userService;
+
+    @PostMapping("/uploadImage")
+    @ResponseBody
```
У тебя получается что это рест контроллер но лежит при этом в папке simple.

tediik replied:
если есть @ResponseBody, при том во всех методах - явно просится переделать на rest-контроллер

tediik commented on src/main/java/com/jm/online_store/controller/rest/AuthorityRestController.java:
```diff
@@ -0,0 +1,35 @@
+                                                @RequestParam String newMail) {
+        User user = (User) auth.getPrincipal();
+        if (userService.isExist(newMail)) {
+            return ResponseEntity.ok("duplicatedEmailError");
```
пишем, что ошибка, но кидаем статус 200?

tediik commented on src/main/java/com/jm/online_store/service/impl/UserServiceImpl.java:
```diff
@@ -84,8 +84,9 @@ public boolean isExist(String email) {
     /**
      * метод добавления нового пользователя.
-     *
+     * <p>
```
это что тут делает? ))

neuromanzer committed:
87-добавление-данных-в-профилелюбом-имя-фамилия-затирает-дефолтный-аватар-смена-пароля-у-того-же-юзера-затирает-ранее-занесенные-в

поправлены замечания с гита

neuromanzer committed:
Merge branch 'dev' into bugfix/user-update

neuromanzer committed:
87-добавление-данных-в-профилелюбом-имя-фамилия-затирает-дефолтный-аватар-смена-пароля-у-того-же-юзера-затирает-ранее-занесенные-в

neuromanzer committed:
87-добавление-данных-в-профилелюбом-имя-фамилия-затирает-дефолтный-аватар-смена-пароля-у-того-же-юзера-затирает-ранее-занесенные-в

tediik commented on src/main/java/com/jm/online_store/controller/rest/AllUsersRestController.java:
```diff
@@ -31,4 +44,18 @@
                 .setRoles(currentUser.getRoles()));
     }
+
+    @PostMapping("/uploadImage")
+    @ResponseBody
+    public String handleImagePost(@RequestParam("imageFile") MultipartFile imageFile) throws IOException {
+        User userDetails = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        return userService.updateUserImage(userDetails.getId(), imageFile);
+    }
+
+    @DeleteMapping("/deleteImage")
+    @ResponseBody
+    public String deleteImage() throws IOException {
+        User userDetails = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        return userService.deleteUserImage(userDetails.getId());
+    }
```
ну мы же не так делаем рест-контроллеры
1. убрать @ResponseBody
2. возвращать ResponseEntity<String> (в этом конкретном твоем случае)
3. соответственно строки с return - переписать с оглядкой на предыдущий пункт

neuromanzer committed:
87-добавление-данных-в-профилелюбом-имя-фамилия-затирает-дефолтный-аватар-смена-пароля-у-того-же-юзера-затирает-ранее-занесенные-в
поправлен AllUsersRestController
